src/lib/modelUtils.ts

```diff
@@ -0,0 +1,335 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { DataTypes, Model } from 'sequelize';
+import * as dotWild from 'dot-wild';
+import merge from 'deepmerge';
+
+// Define the mapping between Sequelize data types and TypeScript data types
+const dataTypeMapping = {
+  [DataTypes.ABSTRACT.key]: undefined,
+  [DataTypes.STRING.key]: 'string',
+  [DataTypes.CHAR.key]: 'string',
+  [DataTypes.TEXT.key]: 'string',
+  [DataTypes.NUMBER.key]: 'number',
+  [DataTypes.TINYINT.key]: 'number',
+  [DataTypes.SMALLINT.key]: 'number',
+  [DataTypes.MEDIUMINT.key]: 'number',
+  [DataTypes.INTEGER.key]: 'number',
+  [DataTypes.BIGINT.key]: 'number',
+  [DataTypes.FLOAT.key]: 'number',
+  [DataTypes.REAL.key]: 'number',
+  [DataTypes.DOUBLE.key]: 'number',
+  [DataTypes.DECIMAL.key]: 'number',
+  [DataTypes.BOOLEAN.key]: 'boolean',
+  [DataTypes.TIME.key]: 'string',
+  [DataTypes.DATE.key]: 'string',
+  [DataTypes.DATEONLY.key]: 'string',
+  [DataTypes.HSTORE.key]: undefined,
+  [DataTypes.JSON.key]: 'object',
+  [DataTypes.JSONB.key]: 'object',
+  [DataTypes.NOW.key]: undefined,
+  [DataTypes.BLOB.key]: undefined,
+  [DataTypes.RANGE.key]: undefined,
+  [DataTypes.UUID.key]: 'string',
+  [DataTypes.UUIDV1.key]: 'string',
+  [DataTypes.UUIDV4.key]: 'string',
+  [DataTypes.VIRTUAL.key]: undefined,
+  [DataTypes.ENUM.key]: 'string',
+  [DataTypes.ARRAY.key]: undefined,
+  [DataTypes.GEOMETRY.key]: 'object',
+  [DataTypes.GEOGRAPHY.key]: 'object',
+  [DataTypes.CIDR.key]: 'string',
+  [DataTypes.INET.key]: 'string',
+  [DataTypes.MACADDR.key]: 'string',
+  [DataTypes.CITEXT.key]: undefined,
+  [DataTypes.TSVECTOR.key]: undefined,
+  // Add more mappings as needed
+};
+
+// Get the column information using the describe method
+const getColumnInformation = async (model: typeof Model) => {
+  const tableDetails = await model.describe();
```

Comment:
do you think this needs a try/catch

Reply:
Sure, what do you propose we do in the catch? Log, call the handleErrors, something else?

Reply:
Depends on whether you think it makes sense to return an empty array or blow up